### DISCUSSION TITLE
Beta-Ti Reconstructions of Cracked Regions

### DISCUSSION THREAD

**Discussion by Savillian** [mtex-toolbox/mtex#913] [Ask Anything] [answered]
Hi everyone,

I'm trying to complete a Beta-Ti reconstruction of a microstructure with a number of cracked regions. The reconstruction process itself proceeds without issue, but the recalculated microstructure consistently glitches out and tries to remove the crack itself.

I've included three screenshots below demonstrating what the Alpha-Ti microstructure looks like and what the Beta-Ti reconstruction produces. Reducing the size of the grains "removed" prior to reconstruction stops the removal of the cracked region, but instead produces a fragmented reconstruction with a number of grains extending through the fracture.

Do you have any tips on processing or overcoming this issue? I suspect the challenge lies with the reconstruction process trying to bridge between the cracked regions. My script is also included after the files.

Thanks in advance!
Alec
Alpha Microstructure
![Alpha2Removal](https://user-images.githubusercontent.com/38051659/112682047-2b8cf100-8e35-11eb-87e4-dcf96c0f91c2.png)
Beta-Ti Reconstruction with grainSize<1 Removed
![ReconstructionTest1](https://user-images.githubusercontent.com/38051659/112682049-2c258780-8e35-11eb-937b-54b352ae05ad.png)
Beta-Ti Reconstruction with grainSize<2 Removed
![ReconstructionTest2](https://user-images.githubusercontent.com/38051659/112682050-2c258780-8e35-11eb-9e93-7a0bbe45864e.png)

[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',2*degree);
ebsd(grains(grains.grainSize<2)) = []; --> Value changed between both reconstruction images.
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',2*degree);
F = halfQuadraticFilter;
F.alpha = 0.5;
ebsd = smooth(ebsd,F,'fill',grains);
[grains] = calcGrains(ebsd,'angle',2*degree);

plot(ebsd('Titanium (Alpha)'),ebsd('Titanium (Alpha)').orientations,'figSize','large')

%% Reconstruction
beta2alpha = orientation.Burger(ebsd('Titanium (Beta)').CS,ebsd('Titanium (Alpha)').CS);
round2Miller(beta2alpha)

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',1.5*degree,...
  'removeQuadruplePoints');

job = parentGrainReconstructor(ebsd, grains, beta2alpha)

job.calcTPVotes('numFit',2)

job.calcParentFromVote('strict', 'minFit',2.5*degree,'maxFit',5*degree,'minVotes',2)

job.calcGBVotes('noC2C');

job.calcParentFromVote('minFit',5*degree)

job.mergeSimilar('threshold',5*degree)

job.mergeInclusions('maxSize',50)

parentEBSD = job.calcParentEBSD;

figure
ipfKey = ipfColorKey(ebsd('Titanium (Beta)').CS);
ipfKey.inversePoleFigureDirection = vector3d.Y;
color = ipfKey.orientation2color(job.parentGrains.meanOrientation);
plot(job.parentGrains, color, 'figSize', 'large')

**Comment by Qhgu** [comment]
In my opinion, the EBSD data with crack could be very complicated, since the discontinuous nature of the bulk material after cracking. I have tried the procedure of MTEX on one of my EBSD data with fatigue crack, but the crack region is fully filled with reconstructed grains after the reconstruction step. I also have no idea how to deal it up to the present.

I find that Travis Skippon and cochranec proposed several codes to deal the notch sample, you can find the link here: https://cochranec.github.io/notch/. After trial, it is weak to deal this problem currently. I expecet some guys could come up with great idea about it.

Regards,
Gu

**Comment by Qhgu** [comment, marked as answer]
Hi,

After reading the documentation, I find a tip to solve the ebsd map with crack.

In the grain reconstruction step, using the following line:

> [grains,ebsd.grainId] = calcGrains(ebsd,**'unitCell'**,'angle',5*degree);

Then you can find that the crack will not be filled by the reconstructed grains. Enjoy it.

Gu.

**Reply by Savillian** [reply]
Hi Gu,

Thanks for finding that, good find! I'll give the addition a try to see if that solves the issue.

Thanks again!
Alec
